src/core/profiling/endoscope_backend.c

```diff
@@ -0,0 +1,587 @@
+#include "src/core/profiling/endoscope_backend.h"
```

Comment:
All suggested changed committed, except using timers_preciseclock.h at this time. Inline is preferred on the clock function.